doc/release/1.15.0-notes.rst

```diff
@@ -36,6 +36,13 @@ Deprecations
   * `np.ma.load`, `np.ma.dump` - these functions already failed on python 3,
     when called with a string.
 
+* Deallocating an array with writeback semantics unresolved is deprecated.
+  Final calls to ``Py_DECREF`` should be proceeded by a call to
+  ``PyArray_ResolveWritebackIfCopy``.
+
+* ``nditer`` use outside of a context manager where the ``nditer`` has writable
+  arrays is deprecated. This situation proves difficult to detect, the
+  ``DeprecationWarning`` may only be emitted when deallocating the array.
```

Comment:
A few more wording nitpicks:

Maybe rephrase as:

```
 * Users of the C-API should call ``PyArrayResolveWriteBackIfCopy`` or ``PyArray_DiscardWritbackIfCopy`` on any array with the ``WRITEBACKIFCOPY` flag set, before the array is deallocated. A deprecation warning will be emitted if not.

 * Users of ``nditer`` should use the nditer object as a context manager anytime one of the iterator operands is writeable, so that numpy can manage temp-buffer-writeback. A deprecation warning will be emitted if a context manager is not used in these cases. Users of the C-API should call ``NpyIter_Close`` before ``NpyIter_Dealloc``.
```

Aside: What about the idea of emitting the deprecation warning on first nditer use, instead of at deallocation time? I think we mentioned it, not sure if we finished discussing that.

Reply:
>Aside: What about the idea of emitting the deprecation warning on first nditer use, instead of at deallocation time?

I did something along those lines in the nditer.close PR #10184. Perhaps the DeprecationWarning at deallocation is sufficient for now, and after this is released we can be more aggressive about checking misuse in the subseqeunt release?